DigitalLearningSolutions.Web/Scripts/trackingSystem/addNewCentreCourseSelectCourse.ts

```diff
@@ -113,23 +111,21 @@ function getCategoryAndTopicFilterByAndUpdateHiddenInputs() {
   const topicFilterElement = <HTMLSelectElement>document.getElementById('CourseTopic');
   const topicFilterValue = topicFilterElement.value;
 
+  updateFilterByHiddenInput(categoryHiddenInputName, categoryFilterValue);
+  updateFilterByHiddenInput(topicHiddenInputName, topicFilterValue);
+
   if (isNullOrEmpty(categoryFilterValue) && isNullOrEmpty(topicFilterValue)) {
-    return null;
+    return '';

```

Comment:
Just to make sure I am understanding - was the issue here that when a user set a drop down to the All Topics/Categories selection, they did not have a value associated (so we can effectively not filter), so these weren't getting updated correctly? Therefore we ensure we always update the hidden inputs with at least an empty string so the filter-by gets updated, and thus the filtering is actually applied?

Reply:
@DanBloxham-sw Yep exactly!